thirdparty/README.md

```diff
@@ -640,6 +642,10 @@ Collection of single-file libraries used in Godot components.
   * Upstream: https://archive.blender.org/wiki/index.php/Dev:Shading/Tangent_Space_Normal_Maps/
   * Version: 1.0 (2011)
   * License: zlib
+- `miniz.{c,h}`
+  * Upstream: https://github.com/richgel999/miniz
+  * Version: 3.0.2 (293d4db1b7d0ffee9756d035b9ac6f7431ef8492, 2023)
+  * License: MIT
```

Comment:
If we're adding miniz as a standalone library, we should probably remove the copy from `thirdparty/basis_universal/encoder/basisu_miniz.h` so we don't have duplicate symbols and potentially mismatched versions of it.

Reply:
I added `miniz` as a last resort after failing, time and time again, to use the `basisu_miniz` amalgamation - it's problematic, but I think I know how to get it to work now, so before doubling down on having a standalone lib I wanna explore using the `#ifdef TOOLS_ENABLED` preprocessor directive to separate inclusion of just the header part in the case of editor builds, and the whole file with definitions in the case of template builds.